### PR TITLE
Merchant index

### PR DESCRIPTION
**Pull request by dcoleman21**
Complete User Stories:
- [x] 38 disables merchant account
- [x] 39 disables merchant items
- [x] 40 enables merchant account
- [x] 41 enables merchant items

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -3,4 +3,22 @@ def show
...
+  def index
+    @merchants = Merchant.all
+  end
+
+  def update
+    merchant = Merchant.find(params[:id])
+    if merchant.status == "enabled"
+      merchant.update_attribute(:status, "disabled")
+      merchant.disable_all_items
+      flash[:success] = "#{merchant.name} is now disabled"
```
Comment (hashmaster3k) [inline app/controllers/admin/merchants_controller.rb:16]: Looking good

File: app/models/merchant.rb [hunk 2]
```diff
@@ -26,4 +26,16 @@ def distinct_cities
     item_orders.distinct.joins(:order).pluck(:city)
   end
 
+  def disable_all_items
+    # Item.where(merchant_id: self.id).update_all(active?: false) #the question mark != creamy
```
Comment (hashmaster3k) [inline app/models/merchant.rb:30]: Heres why you don't label a column with a ? because AR doesn't like it!

File: config/routes.rb [hunk 3]
```diff
@@ -63,5 +63,7 @@
     get '/dashboard', to: 'dashboard#index'
     post '/orders/:order_id/update', to: 'orders#update'
     get '/merchants/:id', to: 'merchants#show'
+    get '/merchants', to: 'merchants#index'
+    patch '/merchants/:id', to: 'merchants#update'
   end
```
Comment (hashmaster3k) [inline config/routes.rb:68]: Way to keep it restful!

File: db/migrate/20200916194032_add_status_to_merchantts.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToMerchantts < ActiveRecord::Migration[5.2]
+  def change
+    add_column :merchants, :status, :string, default: "enabled"
```
Comment (hashmaster3k) [inline db/migrate/20200916194032_add_status_to_merchantts.rb:3]: Your class name has two "t"s in it. Looks like it doesn't affect it after you made the change to line 3.

File: spec/models/merchant_spec.rb [hunk 5]
```diff
@@ -63,5 +63,16 @@
...
+      @meg.disable_all_items
+      expect(@tire.active?).to eq(false)
+    end
+
+    it "enable_all_items" do
+      @meg.disable_all_items
+      expect(@tire.active?).to eq(false)
+      @meg.enable_all_items
+      expect(@tire.active?).to eq(true)
+    end
```
Comment (hashmaster3k) [inline spec/models/merchant_spec.rb:76]: Good job on the model tests.

**Review by hashmaster3k — APPROVED** [review]

File: app/models/merchant.rb (hunk 2 above)
Comment (dcoleman21) [reply app/models/merchant.rb:30]: Isn't that just something?

File: config/routes.rb (hunk 3 above)
Comment (dcoleman21) [reply config/routes.rb:68]: I like resting

File: db/migrate/20200916194032_add_status_to_merchantts.rb (hunk 4 above)
Comment (dcoleman21) [reply db/migrate/20200916194032_add_status_to_merchantts.rb:3]: Ugh fat fingers

**Pull request merged**